lib/gui/app/pages/main/DriveSelector.jsx

```diff
@@ -0,0 +1,243 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const _ = require('lodash')
+const prettyBytes = require('pretty-bytes')
+const propTypes = require('prop-types')
+const React = require('react')
+const driveConstraints = require('../../../../shared/drive-constraints')
+const utils = require('../../../../shared/utils')
+const TargetSelector = require('../../components/drive-selector/target-selector')
+const SvgIcon = require('../../components/svg-icon/svg-icon.jsx')
+const selectionState = require('../../models/selection-state')
+const settings = require('../../models/settings')
+const store = require('../../models/store')
+const analytics = require('../../modules/analytics')
+const exceptionReporter = require('../../modules/exception-reporter')
+
+/**
+ * @summary Get drive title based on device quantity
+ * @function
+ * @public
+ *
+ * @returns {String} - drives title
+ *
+ * @example
+ * console.log(getDrivesTitle())
+ * > 'Multiple Drives (4)'
+ */
+const getDrivesTitle = () => {
+  const drives = selectionState.getSelectedDrives()
+
+  // eslint-disable-next-line no-magic-numbers
+  if (drives.length === 1) {
+    return _.head(drives).description || 'Untitled Device'
+  }
+
+  // eslint-disable-next-line no-magic-numbers
+  if (drives.length === 0) {
+    return 'No targets found'
+  }
+
+  return `${drives.length} Devices`
+}
+
+/**
+ * @summary Get drive subtitle
+ * @function
+ * @public
+ *
+ * @returns {String} - drives subtitle
+ *
+ * @example
+ * console.log(getDrivesSubtitle())
+ * > '32 GB'
+ */
+const getDrivesSubtitle = () => {
+  const drive = selectionState.getCurrentDrive()
+
+  if (drive) {
+    return prettyBytes(drive.size)
+  }
+
+  return 'Please insert at least one target device'
+}
+
+/**
+ * @summary Get drive list label
+ * @function
+ * @public
+ *
+ * @returns {String} - 'list' of drives separated by newlines
+ *
+ * @example
+ * console.log(getDriveListLabel())
+ * > 'My Drive (/dev/disk1)\nMy Other Drive (/dev/disk2)'
+ */
+const getDriveListLabel = () => {
+  return _.join(_.map(selectionState.getSelectedDrives(), (drive) => {
+    return `${drive.description} (${drive.displayName})`
+  }), '\n')
+}
+
+/**
+ * @summary Open drive selector
+ * @function
+ * @public
+ * @param {Object} DriveSelectorService - drive selector service
+ *
+ * @example
+ * openDriveSelector(DriveSelectorService);
+ */
+const openDriveSelector = (DriveSelectorService) => {
+  DriveSelectorService.open().then((drive) => {
```

Comment:
It would have been nice to convert all `.then`s to `async` functions but we can leave it for later.

Reply:
I actually just de-indented all the methods while converting the service to a component and didn't check exactly what each method was doing.

Reply:
Updated @zvin